src/utils/item.ts

```diff
@@ -130,3 +130,87 @@ export const paginationContentFilter = (
   items
     .filter((i) => i.type !== ItemType.FOLDER)
     .filter((i) => !i.settings?.isPinned);
+
+interface Node {
+  children: string[];
+}
+
+interface Tree {
+  [nodeId: string]: Node;
+}
+
+const createMapTree = (data: DiscriminatedItem[]) => {
+  const childrenTreeMap: Tree = {};
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < data.length; i++) {
+    const ele = data[i];
+    const parents = getParentsIdsFromPath(ele.path, { ignoreSelf: true });
+    const lastParent = parents[parents.length - 1];
+    if (lastParent) {
+      if (childrenTreeMap[lastParent]) {
+        childrenTreeMap[lastParent].children = [
+          ...childrenTreeMap[lastParent].children,
+          ele.id,
+        ];
+      } else {
+        childrenTreeMap[lastParent] = { children: [ele.id] };
+      }
+    }
+  }
+  return childrenTreeMap;
+};
+
+type ItemWithChildren = DiscriminatedItem & {
+  children: ItemWithChildren[];
+};
+
+interface TreeNode {
+  [nodeId: string]: ItemWithChildren;
+}
+
+// handle item children tree
+const buildItemsTree = (data: DiscriminatedItem[]) => {
+  const tree: TreeNode = {};
+  if (data.length === 1) {
+    tree[data[0].id] = { ...data[0], children: [] };
+  }
+  const mapTree: Tree = createMapTree(data);
+  const keys = Object.keys(mapTree);
+  const allChildren: string[] = keys.reduce((acc: string[], key: string) => {
+    const node = mapTree[key as keyof Tree];
+    if (node && node.children) {
+      acc.push(...node.children);
+    }
+    return acc;
+  }, []);
+  const rootKeys = keys.filter((key) => !allChildren.includes(key));
```

Comment:
Why do you need to find the rootKeys? I think for a normal player view you only have one root. Can't you define it in the arguments? Let's not take into account the home view.

Reply:
We need to get the first parent, this is the only way to get it even it's only one root